scripts/package

```diff
@@ -38,4 +49,16 @@ if [ ! -d build/system-charts ]; then
     mkdir -p build
     git clone --branch $SYSTEM_CHART_DEFAULT_BRANCH https://github.com/rancher/system-charts build/system-charts
 fi
-TAG=$TAG REPO=${REPO} go run ../pkg/image/export/main.go build/system-charts $IMAGE $AGENT_IMAGE
+
+if [ ! -d build/charts ]; then
+    mkdir -p build
+    git clone --branch $CHART_DEFAULT_BRANCH https://github.com/rancher/charts build/charts
+
+    # Iterate through chart directories and execute callback to extract latest version tgz
+    find build/charts/assets -type d -maxdepth 1 -exec bash -c 'extract_latest_version_tgz {}' \;
+
+    # Remove index to force building a virtual index like system charts
+    rm -f build/charts/index.yaml build/charts/assets/index.yaml
+fi
+
+TAG=$TAG REPO=${REPO} go run ../pkg/image/export/main.go build/system-charts build/charts $IMAGE $AGENT_IMAGE
```

Comment:
I'm assuming this whole PR does actually work, do you know what happens if an image exists in both system-charts and charts ? If so we should unique-ify it, but if thats hard it isn't an immediate need, shouldn't be hard though.

Reply:
We gather all images and append them to a list. Then they all get normalized (Check for non-mirrors, remove duplicates, and sort ) and written to a text file. Since they all get normalized together, this shouldn't be a problem.